cfme/intelligence/chargeback.py

```diff
@@ -3,88 +3,122 @@
-import cfme.web_ui.toolbar as tb
 import cfme.fixtures.pytest_selenium as sel
-from cfme.web_ui import Form, Select, Tree, fill, flash, form_buttons, match_location
-from cfme.web_ui.form_buttons import FormButton
+import cfme.web_ui.accordion as accordion
+from cfme.web_ui import Form, Tree, fill, flash, form_buttons, match_location
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
 from utils.pretty import Pretty
 from utils.update import Updateable
 from utils.version import LOWEST, pick
-from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
+
 
 assignment_tree = Tree("//div[@id='cb_assignments_treebox']/ul")
-tb_select = partial(tb.select, "Configuration")
+
 
 match_page = partial(match_location, controller='chargeback',
                      title='Chargeback')
 
 
 class ChargebackView(BaseLoggedInPage):
     @property
-    def is_displayed(self):
+    def in_chargeback(self):
         return (
             self.logged_in_as_current_user and
             self.navigation.currently_selected == ['Cloud Intel', 'Chargeback'])
 
+    @property
+    def is_displayed(self):
+        return (
+            self.in_chargeback and
+            self.title.text == 'Compute Chargeback Rates')
+
+    configuration = Dropdown('Configuration')
+
+
+class AddComputeChargebackView(ChargebackView):
+    title = Text('#explorer_title_text')
+
+    description = Input(id='description')
+    currency = Select(id='currency')
+
+    @ParametrizedView.nested
+    class fields(ParametrizedView):  # noqa
+        PARAMETERS = ('name',)
+        ROOT = ParametrizedLocator('.//tr[./td[contains(normalize-space(.), {name|quote})]]')
+
+        @cached_property
+        def row_id(self):
+            attr = self.browser.get_attribute(
+                'id',
+                './td/select[starts-with(@id, "per_time_")]',
+                parent=self)
+            return int(attr.rsplit('_', 1)[-1])
+
+        @cached_property
+        def sub_row_id(self):
+            attr = self.browser.get_attribute(
+                'id',
+                './td/input[starts-with(@id, "fixed_rate_")]',
+                parent=self)
+            return int(attr.rsplit('_', 1)[-1])
+
+        per_time = Select(id=ParametrizedString('per_time_{@row_id}'))
+        per_unit = Select(id=ParametrizedString('per_unit_{@row_id}'))
+        start = Input(id=ParametrizedString('start_{@row_id}_{@sub_row_id}'))
+        finish = Input(id=ParametrizedString('finish_{@row_id}_{@sub_row_id}'))
+        fixed_rate = Input(id=ParametrizedString('fixed_rate_{@row_id}_{@sub_row_id}'))
+        variable_rate = Input(id=ParametrizedString('variable_rate_{@row_id}_{@sub_row_id}'))
+        action_add = Button(title='Add a new tier')
+        action_delete = Button(title='Remove the tier')
+
+    add_button = Button(title='Add')
```

Comment:
Functions may be grouped, but widgets should follow either the order as they are defined or as they need to be filled for correct operation.